pyramid_tm/__init__.py

```diff
@@ -27,102 +35,111 @@ def default_commit_veto(request, response):
         return xtm != 'commit'
     return response.status.startswith(('4', '5'))
 
-def text_(s):
-    if isinstance(s, binary_type):
-        try:
-            return s.decode('utf-8')
-        except UnicodeDecodeError:
-            return s.decode('latin-1')
-    return text_type(s)
 
-class AbortResponse(Exception):
+class AbortWithResponse(Exception):
+    """ Abort the transaction but return a pre-baked response."""
     def __init__(self, response):
         self.response = response
 
+
 def tm_tween_factory(handler, registry):
-    old_commit_veto = registry.settings.get('pyramid_tm.commit_veto', None)
-    commit_veto = registry.settings.get('tm.commit_veto', old_commit_veto)
-    activate = registry.settings.get('tm.activate_hook')
-    attempts = int(registry.settings.get('tm.attempts', 1))
-    commit_veto = resolver.maybe_resolve(commit_veto) if commit_veto else None
-    activate = resolver.maybe_resolve(activate) if activate else None
-    annotate_user = asbool(registry.settings.get("tm.annotate_user", True))
-    assert attempts > 0
+    settings = registry.settings
+    maybe_resolve = lambda val: resolver.maybe_resolve(val) if val else None
+    old_commit_veto = settings.get('pyramid_tm.commit_veto', None)
+    commit_veto = settings.get('tm.commit_veto', old_commit_veto)
+    activate_hook = settings.get('tm.activate_hook')
+    commit_veto = maybe_resolve(commit_veto)
+    activate_hook = maybe_resolve(activate_hook)
+    annotate_user = asbool(settings.get('tm.annotate_user', True))
+
+    if 'tm.attempts' in settings:  # pragma: no cover
```

Comment:
I shipped pyramid_retry 0.1 with support for `tm.attempts` but I changed my mind in 0.2 and now it only supports `retry.attempts`. This is so that `tm.attempts` is unused and I can properly warn people here to stop using it.